Gcc 14 fails to build recent FreeOrion

Bug Report
==========
<!--
Please fill in a a meaningful and concise title in the field above.
It helps us to classify the issue even  before reading the body and handling the
issue a bit better.

Examples:
  * Great: "Unable to select fleet in galaxy map"
  * Good: "Unable to click fleet"
  * Not really okay: "Issue with fleets"
  * Horrible: "HELP!!!"
-->


Environment
-----------
<!--
You don't need to provide the information in section when they are not useful in
the context of this issue.  For example a crashing hosted server may not need
information about the graphic card, but maybe depend on the operating system you
use.  When in doubt provide all information you have available.

You can find the FreeOrion version number in the lower right corner of the game
main screen.

When stating the operating system also note the version of the operating system
so, `Windows 8 Pro` instead of `Windows` or `Mac OSX Mountain Lion` instead of
`OSX`.
-->

* **FreeOrion Version**: https://github.com/freeorion/freeorion/commit/864c03a4ff8fbd2674f6559261006024192c09ee
* **Operating System**: Fedora Rawhide
* **Fetched as** <!-- delete lines below that don't apply -->
  * Compiled from source


Description
-----------
<!--
Add a meaningful description of the bug you encountered.

If a screenshot or image helps to describe the issue content feel free to create
one.

Also attach the log files the game creates.  The log files The log files can be
found on
* Windows: %APPDATA%\FreeOrion
* MacOSX: ~/Library/"Application Support"/FreeOrion
* Linux: ${XDG_DATA_HOME:-~/.local/share}/freeorion
-->

```
2024-01-20T23:53:15.9007082Z [ 19%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/System.cpp.o
2024-01-20T23:53:23.4817207Z /freeorion/universe/System.cpp: In member function 'std::string System::ApparentName(int, const Universe&, bool) const':
2024-01-20T23:53:23.4821384Z /freeorion/universe/System.cpp:189:27: error: 'std::string{std::__cxx11::basic_string<char>::_Alloc_hider{((char*)(& EMPTY_STRING.std::__cxx11::basic_string<char>::<anonymous>.std::__cxx11::basic_string<char>::<unnamed union>::_M_local_buf))}, 0, std::__cxx11::basic_string<char>::<unnamed union>{char [16]{0, '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000', '\000'}}}' is not a constant expression
2024-01-20T23:53:23.4824434Z   189 |     constexpr std::string EMPTY_STRING;
2024-01-20T23:53:23.4825093Z       |                           ^~~~~~~~~~~~
2024-01-20T23:53:23.4854413Z /freeorion/universe/System.cpp:189:27: error: 'std::__cxx11::basic_string<char>()' is not a constant expression because it refers to an incompletely initialized variable
```


Expected Result
---------------
<!--
When you enter an issue please add a description of what behaviour you would
expect from the game instead of the issue.
-->

Successfull compilation.

Steps to reproduce
------------------
<!--
If the error only manifests itself after doing a certain number of actions
please add a list of steps to reproduce the bug.  If the error only occurs in
a certain game state please be sure to add a save game to the issue.

To upload save games please pack them into a Zip archive (*.zip) first.
-->
* Install gcc-14
* Configure and build freeorion.

--- a/GG/src/UnicodeCharsets.cpp
+++ b/GG/src/UnicodeCharsets.cpp
@@ -7,6 +7,7 @@
 //! Some Rights Reserved.  See COPYING file or https://www.gnu.org/licenses/lgpl-2.1.txt
 //! SPDX-License-Identifier: LGPL-2.1-or-later
 
+#include <algorithm>
 #include <map>
 #include <GG/UnicodeCharsets.h>
 #include <GG/utf8/checked.h>

